diff --git a/src/win/udp.c b/src/win/udp.c
index 3daa55f..7c015cc 100644
--- a/src/win/udp.c
+++ b/src/win/udp.c
@@ -790,9 +790,9 @@ int uv__udp_set_source_membership6(uv_udp_t* handle,
   memcpy(&mreq.gsr_source, source_addr, sizeof(mreq.gsr_source));
 
   if (membership == UV_JOIN_GROUP)
-    optname = MCAST_JOIN_SOURCE_GROUP;
+    optname = IP_ADD_SOURCE_MEMBERSHIP;
   else if (membership == UV_LEAVE_GROUP)
-    optname = MCAST_LEAVE_SOURCE_GROUP;
+    optname = IP_DROP_SOURCE_MEMBERSHIP;
   else
     return UV_EINVAL;
 
@@ -856,7 +856,7 @@ int uv_udp_set_source_membership(uv_udp_t* handle,
                                           src_addr6,
                                           membership);
   }
-  
+
   err = uv_ip4_addr(source_addr, 0, src_addr4);
   if (err)
     return err;
